### ISSUE TITLE
Tab Indentation works incorrectly when there are multiple cursors

### ISSUE THREAD

**Issue by remixer-dec** [zed-industries/zed#26157]
### Summary

ifs are shifted to the right, breaking python logic

Steps to trigger the problem:
1. Open editor
2. Copy paste provided code
3. Select language as python and have "tab_size": 2 in settings
4. Put multiple cursors at the beginning of the line
5. Try pressing tab
6. See some ifs shifted, some values kept, some shifted twice
 
 <details><summary>Test code</summary>

<pre>
def pca_projection(tensor):
  """
  GPU PCA with explained variance coloring.
  Instead of returning raw principal components, compute a 2D histogram
  (with 128x128 bins) of the first two components for a visible density plot.
  """
  tensor = tensor.float()
  # Reshape: each row is a flattened point.
  data = tensor.reshape(tensor.shape[0], -1).T  # shape: (num_points, features)
  data_centered = data - data.mean(dim=0, keepdim=True)
  _, _, V = torch.linalg.svd(data_centered, full_matrices=False)
  components = data_centered @ V[:, :2]  # shape: (num_points, 2)
  comp_np = components.cpu().numpy()
  # Define histogram range and bins.
  x_min, x_max = comp_np[:, 0].min(), comp_np[:, 0].max()
  y_min, y_max = comp_np[:, 1].min(), comp_np[:, 1].max()
  if x_min == x_max:
    x_min, x_max = x_min - 1, x_max + 1
  if y_min == y_max:
    y_min, y_max = y_min - 1, y_max + 1
  H, _, _ = np.histogram2d(comp_np[:, 0], comp_np[:, 1], bins=128, range=[[x_min, x_max], [y_min, y_max]], density=True)
  H = H.T  # transpose so the orientation is correct
  H_norm = (H - H.min()) / (H.max() - H.min() + 1e-7)
  return torch.tensor(H_norm, dtype=torch.float32)
</pre>
 
</details>

Actual Behavior:

https://github.com/user-attachments/assets/881d6329-6ed0-42f3-a7d3-5898d6a1b5a7

Expected Behavior:

https://github.com/user-attachments/assets/b932ea33-9926-42b6-a077-f3dc4e3cf0fc

 

### Zed Version and System Specs

Zed: v0.174.7 (Zed) 
OS: macOS 14.4.1
Memory: 32 GiB
Architecture: aarch64

**Comment by probably-neb** [comment]
Thanks for reporting, assuming the "Expected Behavior" video was also created in Zed as well, what was the set of actions you used for the correct behavior? I'd like to be able to compare and contrast the working solution with the not working solution to figure out where we're going wrong internally

**Comment by remixer-dec** [comment]
No, expected behavior video was created in vscode, however it does work correctly in Zed when syntax highlighting is set to plain text

**Comment by remixer-dec** [comment]
I still experience this in 0.186.9

**Comment by smitbarmase** [comment]
The fix will be rolling out to the stable version soon. If you're eager to get it right away, please try preview build.

**Comment by remixer-dec** [comment]
updated to 0.188.3, can confirm that this is fixed, thanks!

but I faced another issue (unrelated to this one), not sure if it was ever reported and what even caused it
After a restart, I noticed that among open files, that there are a lot of changes related to code removal that I didn't make.
And these changes look really odd like deleting the definition of a function or deleting a try block without the "except" section, or deleting imports that were actually used. So far this only happened on some open files and these changes are not saved and can be rolled back by closing the file or using git panel, but the fact that it happened and I have no idea why is concerning. 

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/2fa1c99a-01b1-4252-ac9a-9c10502822eb" />

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/d076933f-b778-4aab-9acb-758d5807dcb6" />

<img width="660" alt="Image" src="https://github.com/user-attachments/assets/e4bc8b10-f8ab-4953-afbd-1ac47b16c55c" />

<img width="761" alt="Image" src="https://github.com/user-attachments/assets/25cc1f2f-ffb0-445e-b8af-83cb87ddd284" />

<img width="793" alt="Image" src="https://github.com/user-attachments/assets/d6528cbe-d108-425a-bc71-782bb86586ef" />

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/f186e742-d7d4-4444-b1a7-b1d0aaba6e59" />

**Comment by remixer-dec** [comment]
I'm again facing this bug with multiline indentation in the latest version.
This time with shell scripts.
It adds spaces at the start when I try to add comments on multiple lines

code for testing:
```
  echo "[INFO] Changing directory to $GAMEDIR for gptokeyb"
  cd "$GAMEDIR"
  if [ -f "$GAMEDIR/$gptk_filename" ]; then
    echo "[INFO] Starting gptokeyb2 with keymap $GAMEDIR/$gptk_filename"
    if [ -n "$GPTOKEYB" ]; then
      echo "[INFO] Using GPTOKEYB2 variable from control.txt: $GPTOKEYB2"
      #env LD_PRELOAD="" $GPTOKEYB2 -c "$GAMEDIR/$gptk_filename" &
      #env LD_PRELOAD="$controlfolder/libinterpose.aarch64.so" \
      $GPTOKEYB "java" -c "$GAMEDIR/$gptk_filename" &
      sleep 2
    else
      echo "[WARN] GPTOKEYB variable is not set. Skipping gptokeyb."
    fi
  else
    echo "[WARN] GPTOKEYB keymap file not found: $GAMEDIR/$gptk_filename"
  fi
```
- change language to shell
- select all the lines and move the cursor at the start of the line
- press # 
- see (pic 1) instead of (pic 2)

<img width="204" height="479" alt="Image" src="https://github.com/user-attachments/assets/1db49949-997f-4a0c-af46-a22713306590" />

<img width="147" height="520" alt="Image" src="https://github.com/user-attachments/assets/2aa25998-8d07-4d7d-b45e-4634c1bbaed7" />

**Comment by smitbarmase** [comment]
@remixer-dec I'm moving this to separate issue as it's related to Bash.

**Comment by smitbarmase** [comment]
@remixer-dec fixed for Bash by https://github.com/zed-industries/zed/pull/35116.
